### ISSUE TITLE
Fallback for submodules

### ISSUE THREAD

**Issue by W1M0R** [casey/just#2327]
@casey Thanks for adding submodule support, it's working great.

Currently, when you invoke a submodule recipe within a directory containing a justfile with `set fallback` but no submodules defined (the requested submodule is defined in a parent justfile), the following error is encountered:

```pwsh
❯ just myapp::start       
error: Justfile does not contain submodule `myapp`
```

It would be nice if the `fallback` setting (or another one, e.g. `fallback-submodules`) would allow `just` to look for a requested submodule in parent justfiles.

Sometimes you are in a nested folder structure, and you quickly want to call a recipe defined in the root justfile. Since fallback does not work for submodules, the workaround is to add the submodule to the justfile at your level in the hierarchy (even though the recipe you want to run is unrelated to that justfile), or to cd back to the root and run the recipe from there, or to add a recipe in the root justfile that calls into its own submodule definition.

### Example

This is what happens currently:

```pwsh
❯ cd proj/foo/bar
❯ just --version    
just 1.34.0
❯ just hello
hello
❯ just myapp::start       
error: Justfile does not contain submodule `myapp`
```

This is what should happen:

```pwsh
❯ cd proj/foo/bar
❯ just hello
hello
❯ just myapp::start       
starting
```

#### proj/justfile

```justfile
mod myapp 'justfile-myapp.just'
default:
  just --list
```

#### proj/justfile-myapp.just

```justfile
start:
  @echo "starting"
```

#### proj/foo/bar/justfile

```justfile
set fallback
# set fallback-submodules

hello:
  @echo "hello"
```

**Comment by casey** [comment]
Thanks for the report! Fortunately, this was extremely easy to implement. Fixed in #2329.

**Comment by casey** [comment]
Available in 1.35.0, just released: https://github.com/casey/just/releases/tag/1.35.0

**Comment by W1M0R** [comment]
Thanks @casey!
